### ISSUE TITLE
Doesn't work with AMD modules

### ISSUE THREAD

**Issue by gurobokum** [vercel/ncc#452]
## Subj  
I have subdependency of [@mozilla/source-map](https://www.npmjs.com/package/source-map) which uses AMD modules. During running lambda on zeit.co I get
```
START RequestId: 00e7f507-36c3-4fe1-afcf-528f5a0f88b6 Version: $LATEST
2019-07-06T05:59:17.718Z	00e7f507-36c3-4fe1-afcf-528f5a0f88b6	app
2019-07-06T05:59:17.719Z	00e7f507-36c3-4fe1-afcf-528f5a0f88b6	before app
2019-07-06T05:59:17.954Z	00e7f507-36c3-4fe1-afcf-528f5a0f88b6	TypeError: baseName.split is not a function
    at normalize (/var/task/bin/www:22917:38)
    at stringRequire (/var/task/bin/www:23049:18)
    at amdRequire (/var/task/bin/www:22959:24)
    at Object.<anonymous> (/var/task/bin/www:6739:19)
    at runFactory (/var/task/bin/www:23029:30)
    at define (/var/task/bin/www:23123:13)
    at Object.<anonymous> (/var/task/bin/www:6737:1)
    at __webpack_require__ (/var/task/bin/www:23:30)
    at Object.<anonymous> (/var/task/bin/www:24102:30)
    at __webpack_require__ (/var/task/bin/www:23:30)
END RequestId: 00e7f507-36c3-4fe1-afcf-528f5a0f88b6
REPORT RequestId: 00e7f507-36c3-4fe1-afcf-528f5a0f88b6	Duration: 507.87 ms	Billed Duration: 600 ms 	Memory Size: 3008 MB	Max Memory Used: 86 MB	
RequestId: 00e7f507-36c3-4fe1-afcf-528f5a0f88b6 Process exited before completing request
``` 

As a solution it would be great to have webpack-amd-loader into ncc. out of the box. Or is there any another workaround? 

## TLDR
The issue contains two different sections - [amdefine](https://www.npmjs.com/package/amdefine) + webpack work and AMD loading.

### Amdefine + webpack
[@mozilla/source-map](https://www.npmjs.com/package/source-map) uses [amdefine](https://www.npmjs.com/package/amdefine) as plugin for loading amd modules. [Amdefined throws](https://github.com/jrburke/amdefine/blob/e59edc9da24404ec7937098e3992f8fb0e260be7/amdefine.js#L71) and error when requires webpack module, cause its name is a Number and it throws an error. I've converted it to String and got the second issue cause file isn't copied to fs

### AMD loading
`amd/index.js`
```
if (typeof define !== 'function') {
  var define = require('amdefine')(module, require);
}

define(function (require) {
  const app = require('./app');
});
```
`amd/app.js`
```
if (typeof define !== 'function') {
  var define = require('amdefine')(module, require);
}

define(function (require) {
  return null
})
```
ncc
```
$ (master) ncc run amd/index.js 
ncc: Version 0.20.4
ncc: Compiling file index.js
 13kB  index.js
  2kB  index.js.map
121kB  sourcemap-register.js
134kB  [294ms] - ncc 0.20.4
Error: Cannot find module './app'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:603:15)
    at Function.Module._load (internal/modules/cjs/loader.js:529:25)
    at Module.require (internal/modules/cjs/loader.js:658:17)
    at require (internal/modules/cjs/helpers.js:22:18)
    at stringRequire (/tmp/bcacd817f0cf91cd602898d76ae24167/index.js:275:28)
    at amdRequire (/tmp/bcacd817f0cf91cd602898d76ae24167/index.js:165:24)
    at Object.<anonymous> (/tmp/bcacd817f0cf91cd602898d76ae24167/index.js:370:15)
    at runFactory (/tmp/bcacd817f0cf91cd602898d76ae24167/index.js:237:30)
    at define (/tmp/bcacd817f0cf91cd602898d76ae24167/index.js:332:13)
    at Object.350 (/tmp/bcacd817f0cf91cd602898d76ae24167/index.js:369:1)
```

**Comment by styfle** [comment]
Closing since there are no plans to support AMD modules
